### PR TITLE
Add configurable logger; add getNextState method

### PR DESCRIPTION
**Pull request by f3nrir92**
Sometimes it's useful to know what state will be next without actually dispatching an event.

Along with i've added ability to pass custom logger. This is useful when collecting logs from stdout in JSON format (in this case output from console's methods will be unwanted)

### CODE REVIEW

File: src/stateMachine.ts [hunk 1]
```diff
@@ -49,6 +65,21 @@ export class StateMachine<STATE, EVENT extends string | number | symbol, CALLBAC
     return this.transitions.some((trans) => (trans.fromState === this._current && trans.event === event));
   }
 
+  async getNextState(event: EVENT): Promise<STATE> {
```
Comment (eram) [inline src/stateMachine.ts:68]: there's no reason to make it an async function. also no reason to log an error.

```
   getNextState(event: EVENT): STATE | undefined {
    const transition = this.transitions.find((tran) => tran.fromState === this._current && tran.event === event);
    return transition?.toState;
  }

```

File: src/__test__/stateMachine.test.ts [hunk 2]
```diff
@@ -108,6 +108,7 @@ describe("stateMachine tests", () => {
     expect(door.isOpen()).toBeFalsy();
     expect(door.isBroken()).toBeFalsy();
     expect(door.can(Events.open)).toBeTruthy();
+    expect(await door.getNextState(Events.open)).toEqual(States.opening);
```
Comment (eram) [inline src/__test__/stateMachine.test.ts:111]: fix according to below

File: src/__test__/stateMachine.test.ts [hunk 3]
```diff
@@ -117,6 +118,12 @@ describe("stateMachine tests", () => {
     const door = new Door(undefined, States.opened);
     expect(door.can(Events.open)).toBeFalsy();
 
+    door.getNextState(Events.open).then(() => {
```
Comment (eram) [inline src/__test__/stateMachine.test.ts:121]: fix according to below

**Review by eram — CHANGES_REQUESTED** [review]

File: src/stateMachine.ts (hunk 1 above)
Comment (f3nrir92) [reply src/stateMachine.ts]: Fixed

**Review by eram — APPROVED** [review]

**Pull request merged**